callouts/python/extproc/example/ext_proc_client.py

```diff
@@ -28,18 +28,31 @@
 )
 
 
+def _make_channel(
+  address: tuple[str, int], key: str | None = None
+) -> grpc.Channel:
+  addr_str = _addr_to_str(address)
+  if key:
+    with open(key, 'rb') as file:
+      creds = grpc.ssl_channel_credentials(file.read())
+      return grpc.secure_channel(addr_str, creds)
+  else:
+    return grpc.insecure_channel(addr_str)
+
+
 def make_json_request(
-  json_list: list[str], address: tuple[str, int]
+  json_list: list[str], address: tuple[str, int], key: str | None = None
 ) -> Iterator[ProcessingResponse]:
   """Make requests to a callout service with ProcessingRequest json data.
   Args:
     json_list: A list of json strings representing ProcessingRequest data.
     address: ip address of the callout server.
+    key: Local filepath to a chain authenication certificate.
   Returns:
     An iterator containg each response.
   """
   callouts = [Parse(data, ProcessingRequest()) for data in json_list]
-  with grpc.insecure_channel(_addr_to_str(address)) as channel:
+  with _make_channel(address, key) as channel:
     stub = ExternalProcessorStub(channel)
     for response in stub.Process(iter(callouts)):
```

Comment:
So the user could have a series of request_headers as messages? Will they be able to exercise how their callout server handles stream closure given that envoy creates a new stream for each individual client request?

Reply:
The requests for this client should be able to handle individual stream closures, if the server closes one request it will close all requests though. The user would need to make multiple calls to the docker binary if they want to test multiple closures.